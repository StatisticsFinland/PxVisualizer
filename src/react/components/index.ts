export * from "./chart";
export * from "./ErrorBoundary";
export * from "./globalStyle";
export * from "./icon";
export * from "./burgerMenu";
