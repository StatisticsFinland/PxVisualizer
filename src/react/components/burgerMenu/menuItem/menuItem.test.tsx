import { render, screen } from "@testing-library/react";
import MenuItem from "./menuItem";
import React from "react";
import 'jest-styled-components';

const mockFunction = jest.fn();

const mockImg = (
    <img src={'https://www.google.fi/favicon.ico'} />
);

describe('MenuItem, rendering tests', () => {
    it('should render', () => {
        const {asFragment} = render(<MenuItem locale={'fi'} text={'NAPPI'} onClick={mockFunction} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render with correct prefix icon', () => {
        const {asFragment} = render(<MenuItem prefixIcon={'Download'} locale={'fi'} text={'NAPPI'} onClick={mockFunction} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render with correct suffix icon', () => {
        const {asFragment} = render(<MenuItem suffixIcon={'Download'} locale={'fi'} text={'NAPPI'} onClick={mockFunction} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render with correct external icon', () => {
        const {asFragment} = render(<MenuItem isExternal={true} locale={'fi'} text={'NAPPI'} url={'foobar'} />);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render with correct custom suffix icon', () => {
        const {asFragment} = render(<MenuItem suffixIcon={mockImg} locale={'fi'} text={'NAPPI'} onClick={mockFunction} />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('MenuItem, functional tests', () => {
    it('should call onclick function when clicked', () => {
        const btnText = 'PRESSME';
        render(<MenuItem locale={'fi'} text={btnText} onClick={mockFunction} />);
        screen.getByText(btnText).click();
        expect(mockFunction).toBeCalledTimes(1);
        screen.getByText(btnText).click();
        expect(mockFunction).toBeCalledTimes(2);
    });

    it('should render a link', () => {
        const btnText = 'PRESSME';
        render(<MenuItem locale={'fi'} text={btnText} url={'foobar.fi'} openNewTab={true} />);
        expect(screen.getAllByRole('link').length).toBe(1);
    });

    it('should render nothing if functionality is not declared', () => {
        const btnText = 'PRESSME';
        render(<MenuItem locale={'fi'} text={btnText} />);
        expect(screen.queryByText(btnText)).toBeNull();
    });
});