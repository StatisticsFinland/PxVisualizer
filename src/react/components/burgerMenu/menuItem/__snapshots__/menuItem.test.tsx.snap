// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem, rendering tests should render 1`] = `
<DocumentFragment>
  .c1 {
  background-color: transparent;
  text-align: left;
  border: none;
  margin: 0;
  padding: 0.5rem 0.7rem 0.5rem 0.7rem;
  cursor: pointer;
  display: inline;
  font-family: "Barlow Semi Condensed",Verdana,sans-serif;
  min-width: 14rem;
  border-bottom: 0px;
  color: #1a3061;
  min-height: 3rem;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  border-top-right-radius: 0px;
  border-top-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
}

.c0:hover {
  background-color: #f2f2f2;
}

.c2 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

.c3 {
  margin-left: 0px;
  margin-right: 0px;
}

<li
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          NAPPI
        </span>
      </div>
    </button>
  </li>
</DocumentFragment>
`;

exports[`MenuItem, rendering tests should render with correct custom suffix icon 1`] = `
<DocumentFragment>
  .c1 {
  background-color: transparent;
  text-align: left;
  border: none;
  margin: 0;
  padding: 0.5rem 0.7rem 0.5rem 0.7rem;
  cursor: pointer;
  display: inline;
  font-family: "Barlow Semi Condensed",Verdana,sans-serif;
  min-width: 14rem;
  border-bottom: 0px;
  color: #1a3061;
  min-height: 3rem;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  border-top-right-radius: 0px;
  border-top-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
}

.c0:hover {
  background-color: #f2f2f2;
}

.c2 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

.c4 {
  height: 1rem;
  width: 1rem;
}

.c4 svg {
  height: 1rem;
  width: 1rem;
  fill: #1a3061;
}

.c3 {
  margin-left: 0px;
  margin-right: 0.5rem;
}

<li
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          NAPPI
        </span>
        <div
          class="c4"
        />
      </div>
    </button>
  </li>
</DocumentFragment>
`;

exports[`MenuItem, rendering tests should render with correct external icon 1`] = `
<DocumentFragment>
  .c2 {
  background-color: transparent;
  text-align: left;
  border: none;
  margin: 0;
  padding: 0.5rem 0.7rem 0.5rem 0.7rem;
  cursor: pointer;
  display: inline;
  font-family: "Barlow Semi Condensed",Verdana,sans-serif;
  min-width: 14rem;
  border-bottom: 0px;
  color: #1a3061;
  min-height: 3rem;
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  border-top-right-radius: 0px;
  border-top-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
}

.c0:hover {
  background-color: #f2f2f2;
}

.c3 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

.c6 {
  height: 1rem;
  width: 1rem;
}

.c6 svg {
  height: 1rem;
  width: 1rem;
  fill: #1a3061;
}

.c4 {
  margin-left: 0px;
  margin-right: 0.5rem;
}

.c5 {
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  -webkit-clip-path: inset(50%);
  clip-path: inset(50%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

<li
    class="c0"
  >
    <a
      class="c1"
      href="foobar"
    >
      <button
        class="c2"
      >
        <div
          class="c3"
        >
          <span
            class="c4"
          >
            NAPPI
          </span>
          <span
            class="c5"
          >
            Ulkoinen linkki
          </span>
          <div
            class="c6"
          >
            <svg
              aria-hidden="true"
              height="79.62mm"
              viewBox="0 0 251.84 225.69"
              width="88.84mm"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m125.89,225.68c-22.84,0-45.67.03-68.51,0C23.18,225.63.03,202.4.01,168.14,0,131.14,0,94.13.01,57.13.04,23.53,23.35.14,56.85.1c23.67-.03,21.34.02,45.01-.03,4.75,0,8.96,1.04,11.87,5.09,2.64,3.67,3.08,7.7,1.25,11.83-2.01,4.53-5.59,7-10.57,7.06-9,.11-12,.07-21,.07-15.84,0-11.67-.05-27.5.02-18.82.08-31.81,12.98-31.84,31.74-.06,38-.06,76.01,0,114.01.03,18.74,13.04,31.7,31.86,31.72,46.67.05,93.34.06,140.01,0,18.79-.02,31.71-12.99,31.79-31.8.07-17,0-14,.03-31.01.02-8.3,4.75-13.75,11.83-13.81,7.07-.06,12.12,5.48,12.16,13.61.07,17.5.18,15.01-.02,32.51-.34,30.71-23.87,54.02-54.81,54.56h-71.01l-.02.02Z"
              />
              <path
                d="m207.19,24.08c-6.95,0-23.9.04-30.85,0-8.12-.06-13.27-4.55-13.44-11.6-.17-7.19,5.18-12.33,13.32-12.38,17.33-.11,44.67-.12,62,0,8.48.06,13.45,4.99,13.5,13.37.12,16.83.12,43.67,0,60.5-.06,7.69-5.4,13.04-12.34,12.87-6.83-.17-11.53-5.35-11.61-12.97-.08-7.63-.02-25.26-.02-34.12-2.09,1.8-3.42,2.88-4.68,4.03-26.76,24.57-53.52,49.15-80.28,73.72-1.6,1.47-3.13,3-4.82,4.36-5.57,4.47-12.54,3.94-16.98-1.2-4.31-4.98-4.06-11.78,1.06-16.6,10.92-10.28,22.03-20.36,33.07-30.51,16.56-15.23,33.13-30.45,49.72-45.65.97-.89,2.15-1.56,3.23-2.33-.29-.5-.58-.99-.87-1.49h0Z"
              />
            </svg>
          </div>
        </div>
      </button>
    </a>
  </li>
</DocumentFragment>
`;

exports[`MenuItem, rendering tests should render with correct prefix icon 1`] = `
<DocumentFragment>
  .c1 {
  background-color: transparent;
  text-align: left;
  border: none;
  margin: 0;
  padding: 0.5rem 0.7rem 0.5rem 0.7rem;
  cursor: pointer;
  display: inline;
  font-family: "Barlow Semi Condensed",Verdana,sans-serif;
  min-width: 14rem;
  border-bottom: 0px;
  color: #1a3061;
  min-height: 3rem;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  border-top-right-radius: 0px;
  border-top-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
}

.c0:hover {
  background-color: #f2f2f2;
}

.c2 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

.c3 {
  height: 1rem;
  width: 1rem;
}

.c3 svg {
  height: 1rem;
  width: 1rem;
  fill: #1a3061;
}

.c4 {
  margin-left: 0.5rem;
  margin-right: 0px;
}

<li
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <svg
            aria-hidden="true"
            height="33.64mm"
            viewBox="0 0 95.35 95.37"
            width="33.64mm"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m41.74,51.99c0-1.25,0-1.92,0-2.59,0-14.4,0-28.79,0-43.19,0-3.27,2.24-5.84,5.28-6.17,3.01-.33,5.8,1.65,6.48,4.66.16.71.14,1.48.14,2.22,0,14.15,0,28.29,0,42.44v2.34c.53-.42.91-.67,1.23-.98,3.12-3.11,6.23-6.23,9.35-9.34,2.74-2.72,6.26-2.87,8.75-.4,2.49,2.46,2.39,6.06-.3,8.75-6.79,6.81-13.59,13.61-20.4,20.4-2.92,2.91-6.25,2.92-9.15.03-6.81-6.79-13.61-13.59-20.4-20.4-2.71-2.72-2.84-6.26-.35-8.75,2.48-2.49,6.03-2.36,8.75.34,3.42,3.4,6.82,6.83,10.61,10.64Z"
            />
            <path
              d="m47.67,95.37c-13.83,0-27.66,0-41.49,0-3.18,0-5.7-2.16-6.12-5.13-.42-2.98,1.5-5.82,4.51-6.59.65-.17,1.35-.18,2.02-.18,27.41,0,54.83,0,82.24,0,3.8,0,6.48,2.43,6.52,5.88.04,3.53-2.66,6.02-6.56,6.03-13.71,0-27.41,0-41.12,0Z"
            />
          </svg>
        </div>
        <span
          class="c4"
        >
          NAPPI
        </span>
      </div>
    </button>
  </li>
</DocumentFragment>
`;

exports[`MenuItem, rendering tests should render with correct suffix icon 1`] = `
<DocumentFragment>
  .c1 {
  background-color: transparent;
  text-align: left;
  border: none;
  margin: 0;
  padding: 0.5rem 0.7rem 0.5rem 0.7rem;
  cursor: pointer;
  display: inline;
  font-family: "Barlow Semi Condensed",Verdana,sans-serif;
  min-width: 14rem;
  border-bottom: 0px;
  color: #1a3061;
  min-height: 3rem;
}

.c1:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c0 {
  border-top-right-radius: 0px;
  border-top-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-left-radius: 0px;
}

.c0:hover {
  background-color: #f2f2f2;
}

.c2 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: center;
}

.c4 {
  height: 1rem;
  width: 1rem;
}

.c4 svg {
  height: 1rem;
  width: 1rem;
  fill: #1a3061;
}

.c3 {
  margin-left: 0px;
  margin-right: 0.5rem;
}

<li
    class="c0"
  >
    <button
      class="c1"
    >
      <div
        class="c2"
      >
        <span
          class="c3"
        >
          NAPPI
        </span>
        <div
          class="c4"
        >
          <svg
            aria-hidden="true"
            height="33.64mm"
            viewBox="0 0 95.35 95.37"
            width="33.64mm"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m41.74,51.99c0-1.25,0-1.92,0-2.59,0-14.4,0-28.79,0-43.19,0-3.27,2.24-5.84,5.28-6.17,3.01-.33,5.8,1.65,6.48,4.66.16.71.14,1.48.14,2.22,0,14.15,0,28.29,0,42.44v2.34c.53-.42.91-.67,1.23-.98,3.12-3.11,6.23-6.23,9.35-9.34,2.74-2.72,6.26-2.87,8.75-.4,2.49,2.46,2.39,6.06-.3,8.75-6.79,6.81-13.59,13.61-20.4,20.4-2.92,2.91-6.25,2.92-9.15.03-6.81-6.79-13.61-13.59-20.4-20.4-2.71-2.72-2.84-6.26-.35-8.75,2.48-2.49,6.03-2.36,8.75.34,3.42,3.4,6.82,6.83,10.61,10.64Z"
            />
            <path
              d="m47.67,95.37c-13.83,0-27.66,0-41.49,0-3.18,0-5.7-2.16-6.12-5.13-.42-2.98,1.5-5.82,4.51-6.59.65-.17,1.35-.18,2.02-.18,27.41,0,54.83,0,82.24,0,3.8,0,6.48,2.43,6.52,5.88.04,3.53-2.66,6.02-6.56,6.03-13.71,0-27.41,0-41.12,0Z"
            />
          </svg>
        </div>
      </div>
    </button>
  </li>
</DocumentFragment>
`;
